pkg/iam/policy/actionset.go

```diff
@@ -99,6 +99,11 @@ func (actionSet *ActionSet) UnmarshalJSON(data []byte) error {
 	for _, s := range sset.ToSlice() {
 		action, err := parseAction(s)
 		if err != nil {
+			adminAction, aerr := parseAdminAction(s)
+			if aerr == nil {
```

Comment:
let's return a composite error parsing both situations? @poornas - I would suggest we try to parseAdminAction inside parseAction() itself as a first function.